biz.aQute.bndlib/src/aQute/bnd/service/result/Result.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2016 René Perschon <rperschon85@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package aQute.bnd.service.result;
```

Comment:
Isn't this whole package just a Promise? and object holding a result or a failure with methods for monadic mutation?

Reply:
I think promise imply parallel processing, this is just a more advanced Optional. It uses the same pattern as Promise to carry the error forward but so does Option. I really do not see how Promises would fit here, also because they mandate Exceptions for errors. 

I am ok to discuss in the group but I kind of like them. A lot of code gets very clean and I can get back to having exceptions really be exceptions.